core/src/main/java/com/example/core/repository/FirebaseRepositoryImpl.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.core.repository
+
+import android.util.Log
+import com.google.firebase.auth.FirebaseAuth
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.tasks.await
+
+class FirebaseRepositoryImpl: FirebaseRepository {
+
+    override fun createNewUser(email: String, password: String, auth: FirebaseAuth): Flow<Boolean> = callbackFlow {
+
+        Log.d("CreatingUser", "Email: $email, password $password")
+
+        auth.createUserWithEmailAndPassword(email, password).addOnCompleteListener { task ->
+
+            if (task.isSuccessful) {
+                Log.d("CreatingUser", "createUserWithEmail:success")
+                trySend(task.isSuccessful)
+            } else {
+                Log.d("CreatingUser", "createUserWithEmail: failure, ${task.exception}")
+                trySend(task.isSuccessful)
+            }
+        }
+        awaitClose { }
```

Comment:
Please remove logs, and use only one 
trySend(task.isSuccessful)
outside if statement